include/service/entities/plugin_service.js

```diff
@@ -1575,6 +1590,19 @@ PluginService.validateDetails = function(details, pluginDirName, cb) {
 	if (!v.validateVersionNum(details.version, true)) {
 		errors.push("An invalid version number ["+details.version+"] was provided.  Must match the form: xx.xx.xx");
 	}
+    
+    //validate pbVersion
+    if (!v.isVersionExpression(details.pbVersion, false)) {
+      errors.push("An invalid version expression ["+details.pbVersion+"] was provided.");
+    }
+
+    //validate pbVersion in config against pb version
+    if(details.pbVersion){
+    	if(!semver.satisfies(details.pbVersion, pb.config.version)) {
+      		errors.push("The version of ["+details.name+"] ("+ details.pbVersion +") does not match the pencilblue version (" + pb.config.version + ") you are using.");
+    	}
+    }
```

Comment:
This block should be updated to: 
```
//validate pbVersion
    if (pb.utils.isString(details.pbVersion)) {
        
        if (!v.isVersionExpression(details.pbVersion, true)) {
            errors.push('An invalid version expression "'+details.pbVersion+'" was provided.');
        }

        //validate pbVersion in config against pb version
        else if(!semver.satisfies(pb.config.version, details.pbVersion)) {
            errors.push("Version "+details.version+" is incompatible with PencilBlue version "+pb.config.version);
        }
    }
```

These changes are necessary because we want to verify that a value exists before do any other tests.  Next, the second if statement becomes an "else if" because if the first error's we know the second will fail.  The next change we make is to the call to satisfies.  The two arguments are in reverse order.